JavaScriptSPA/authPopup.js

```diff
@@ -0,0 +1,58 @@
+// Create the main myMSALObj instance
+// configuration parameters are located at authConfig.js
+const myMSALObj = new Msal.UserAgentApplication(msalConfig);
+
+function signIn() {
+  myMSALObj.loginPopup(loginRequest)
+    .then(loginResponse => {
+      console.log(loginResponse);
+      console.log('id_token acquired at: ' + new Date().toString());
+      if (myMSALObj.getAccount()) {
+        showWelcomeMessage(myMSALObj.getAccount());
+      }
+    }).catch((error) => {
+      console.log(error);
+    });
+}
+
+function signOut() {
+  myMSALObj.logout();
+}
```

Comment:
Anything like this that is the same across both files should be factored into a single place, instead of being duplicated, in my opinion.

Reply:
I know you have a valid point but I couldn't figure out where else this would fit logically. To keep the convenience of just switching authPopup with authRedirect, this part above should either go to a new js file or to the index.html, neither is appealing... @sameerag